contentgrid-spring-data-support/src/test/java/com/contentgrid/data/support/auditing/v1/AuditorAwareImplTest.java

```diff
@@ -0,0 +1,304 @@
+                        .with(john))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void getEntity_shouldCheckIfUnmodifiedSince() throws Exception {
+        var response = mockMvc.perform(post("/invoices")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "123456"}
+                                """)
+                        .with(john))
+                .andExpect(status().isCreated())
+                .andReturn();
+        var invoiceId = StringUtils.substringAfterLast(response.getResponse().getHeader("Location"), "/");
+        var dateAfterCreation = Instant.now();
+        var dateBeforeCreation = dateAfterCreation.minus(1, ChronoUnit.HOURS);
+
+        mockMvc.perform(get("/invoices/{id}", invoiceId)
+                        .header(HttpHeaders.IF_UNMODIFIED_SINCE, formatInstant(dateAfterCreation))
+                        .with(john))
+                .andExpect(status().isOk());
+        mockMvc.perform(get("/invoices/{id}", invoiceId)
+                        .header(HttpHeaders.IF_UNMODIFIED_SINCE, formatInstant(dateBeforeCreation))
+                        .with(john))
+                .andExpect(status().isPreconditionFailed());
+    }
+
+    @Test
+    @Disabled("ACC-1220")
+    void putEntity_shouldCheckIfUnmodifiedSince() throws Exception {
+        var response = mockMvc.perform(post("/invoices")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "123456"}
+                                """)
+                        .with(john))
+                .andExpect(status().isCreated())
+                .andReturn();
+        var invoiceId = StringUtils.substringAfterLast(response.getResponse().getHeader("Location"), "/");
+        var dateAfterCreation = Instant.now();
+        var dateBeforeCreation = dateAfterCreation.minus(1, ChronoUnit.HOURS);
+
+        mockMvc.perform(put("/invoices/{id}", invoiceId)
+                        .header(HttpHeaders.IF_UNMODIFIED_SINCE, formatInstant(dateBeforeCreation))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "000000"}
+                                """)
+                        .with(bob))
+                .andExpect(status().isPreconditionFailed());
+        mockMvc.perform(put("/invoices/{id}", invoiceId)
+                        .header(HttpHeaders.IF_UNMODIFIED_SINCE, formatInstant(dateAfterCreation))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "000001"}
+                                """)
+                        .with(bob))
+                .andExpect(status().isNoContent());
+    }
```

Comment:
These behaviors are not part of `AuditorAware`, but belongs in contentgrid-spring-data-rest tests